### PR TITLE
implements GetMachineClientForCustomerCluster method

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: adds `ImpersonationConfig` to` restclient.Config`, user information will be taken from `provider.UserInfo` struct. `GetMachineClientForCustomerCluster` is not used because there is no controller in the cluster that would generate proper permissions.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # 2680

**Release note**:

```
NONE
```

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
/hold 

waiting for https://github.com/kubermatic/kubermatic/pull/2684

File: api/pkg/handler/node_v1.go [hunk 1]
```diff
@@ -58,10 +58,8 @@ func deleteNodeForClusterLegacy(projectProvider provider.ProjectProvider) endpoi
 			return nil, common.KubernetesErrorToHTTPError(err)
 		}
 
-		// TODO:
-		// normally we have project, user and sshkey providers
-		// but here we decided to use machineClient and kubeClient directly to access the user cluster.
-		//
+		// TODO: change to
+		// machineClient, err := clusterProvider.GetMachineClientForCustomerCluster(userInfo, cluster)
```
Comment (p0lyn0mial) [inline api/pkg/handler/node_v1.go:62]: once we have this pull all we have to do is to change `GetAdminMachineClientForCustomerCluster` to `GetMachineClientForCustomerCluster` in the HTTP endpoints.

**Comment by p0lyn0mial** [comment]
/hold cancel

@zreigz PTAL

**Review by zreigz — APPROVED** [review]
/lgtm
/approve

**Pull request merged**